packages/lab/src/date-input/DateInput.tsx

```diff
@@ -115,272 +119,285 @@ export interface DateInputProps
    */
   selectionVariant?: "default" | "range";
   /**
-   * Callback fired when the selected daate change.
+   * Callback fired when the selected date change.
    */
   onSelectionChange?: (
     event: SyntheticEvent,
-    selectedDate?: DateValue | { startDate?: DateValue; endDate?: DateValue }
+    selectedDate?: SelectionVariantType
+  ) => void;
+  onChange?: (
```

Comment:
Is it possible to type this so that when single, only 1 value?
e.g. using conditional type depends on `SelectionVariantType`